### PR TITLE
Add excludes to sources.kubernetes

### PR DESCRIPTION
**Pull request by flah00**
* Following up on #17, add exclusions found in sources.container

### CODE REVIEW

**Comment by frankreno** [comment]
thanks for this, I will review it later today.

**Comment by frankreno** [comment]
@flah00 : First off thank you for the pull request, we always appreciate the help.  After thinking about this more, I think there are a few things we should do.    The sources.containers.conf uses a regex pattern primarily because there will be multiple fail paths under the containers log directory and you may want to exclude some but not all.  

With the kubernetes sources however these are all concrete paths.  To avoid unnecessary processing of the logs in fluentd I think we should take advantage of the exclude path option in the fluentd plugin. We currently support this in containers by specifying the EXCLUDE_PATH environment variable.

exclude_path "#{ENV['EXCLUDE_PATH']}"

I think we should be able to to do the same thing here, but with all the sources in the kubernetes.sources.conf.  The advantage of this is being able to exclude logs that are not running as a pod and we do not send the logs to fluentd at all.  Do you want to take a stab at that approach?  I realize my comments in #17 likely led you to this path and after thinking about my comments, I think it was not the best direction so apologies for that.

**Comment by flah00** [comment]
@frankreno Thanks for the detailed explanation. I can't believe I excluded `EXCLUDE_PATH`! If you don't mind, I'm going to update the README, so it explicitly states what each exclude can affect.

**Comment by flah00** [comment]
Just so I'm clear, the `EXCLUDE_PATH` env var is dependent on the absolute path of a given log file. If I wanted to exclude all of the kubernetes log files, I would set the env var to `EXCLUDE_PATH="[\"/mnt/log/*.log\"]"`. If I wanted to only exclude the api server, I would set it to `EXCLUDE_PATH="[\"/mnt/log/kube-apiserver.log\"]"`. If I wanted to exclude the docker logs, I would set it to `EXCLUDE_PATH="[\"/var/lib/docker.log\"]"`.

**Comment by flah00** [comment]
One final thought, it seems like it might also be sensible to include `exclude_namespace` in sources.kubernetes. If I wanted to exclude all of the kube-system namespace I'd be surprised to find out I had to use `exclude_path` instead.

**Comment by frankreno** [comment]
@flah00 : totally fine to update the readme, thank you!  You are correct on the EXCLUDE_PATH, and I believe the tail plugin can accept a comma separated list.   The reason I like this approach is just to save unnecessary processing in fluentd.  With regards to adding the namespace, that is completely sensible and I agree with you.  Thank you again for your help on this.

File: conf.d/source.kubernetes.conf [hunk 1]
```diff
@@ -14,6 +15,7 @@
   source_category salt
   source_name k8s_salt
   source_category_prefix "#{ENV['SOURCE_CATEGORY_PREFIX']}"
+	exclude_namespace_regex "#{ENV['EXCLUDE_NAMESPACE_REGEX']}"
```
Comment (frankreno) [inline conf.d/source.kubernetes.conf]: minor formatting issue here

Comment (flah00) [inline conf.d/source.kubernetes.conf]: I'm not sure what to do about this. Locally, it shows my indent correctly. On github, it doesn't. :/
screen shot 2017-04-06 at 2 48 03 pm

Comment (flah00) [inline conf.d/source.kubernetes.conf]: It's a derpy sorta day.

**Comment by frankreno** [comment]
@flah00 thank you so much for this.   I will update when it is pushed to our Docker Hub.

**Pull request merged**